docs/RFC 001.md

```diff
@@ -0,0 +1,206 @@
+  - Example: Runways may contain the fields:
+    - `ident: "RW23L"`
+    - `airport_ident: "NZAA"`
+  - Example: Airports may contain the field:
+    - `ident: "NZAA"`
+- Using the term `icao` to refer to an identifier is an anti-pattern as it may be confused with the `icao_code` which contains a two-letter code such as `ES` which represents the area of the world where this piece of data lies.
+  - Example: Airports may contain the fields:
+    - `icao_code: "NZ"`
+    - `ident: "NZAA"`
+- Fields should represent the unit that the data will be using a type alias, not by a suffix or prefix to the name of the field.
+  - Do: `runway_length: Feet`
+  - Don't: `runway_length_ft: number`
+- Radio navigation aid data type names should consist of the acronym of the type, followed by the suffix `Navaid`
+  - Examples: `VhfNavaid`, `IlsNavaid`, `NdbNavaid`,
+- Acronyms should **_not_** be all capitalised
+  - Do: `Vhf`
+  - Dont: `NDB`
+- Latitudes should be encoded to as `lat` and Longitudes should be encoded to as `long`, and should wherever they are used in conjunction with each other, be part of a `Coordinates` data structure
+
+---
+
+- Items which are linked to other data should **_not_** have the linked data queried automatically and added to the returned data.
+  - This is because increasing the size of data this much will result in slower queries due to the performance issues with Coherent's JSON parser.
+  - While it is useful to have linked data such as Navaid information on airway or procedures, this would not be viable when using the API in a language other than JS or Rust
+- For items such as Procedures and Airways, which are represented in the database as a large number of rows with the same `ident` and possibly `icao_code`. They should be grouped into an object, containing list(s) which stores the individual elements. Fields which are known to be the same throughout the grouping should be on the group object, not the individual elements.
+  - If the individual elements are split into sections such as transitions in a procedure, these should be encoded as subgroups.
+
+---
+
+- Data should be provided in the same units that the Database provides for consistency with aviation standards
+- Enums should be used where possible and should have values which match the database encoding.
+  - Except for when an enum is to be used as a filter, where its values should support bitwise flags.
+- The data provided by this API should be able to be encoded purely in JSON, meaning:
+
+  - No functions
+  - No classes
+  - This is to support JSON serialization as a form of copying or sending between instruments
+
+- Data fields should never be empty strings, they should be undefined
+- Data fields which are linked to each other and will either both be undefined or both be defined
+- Data being encoded as possibly undefined should be based on database sweeps to find whether there are any null fields, not the database schema.

```

Comment:
The database has no consistency in terms of fields being NOT NULL, many tables have all of their fields as possibly NULL

Reply:
In the discussion over on Slack I got the impression that the columns are missing this restriction on purpose since they cannot be guaranteed to have a value? How can you tell cases like these apart from columns that could very well be defined as NOT NULL?